### ISSUE TITLE
Rustflag unification seems to be broken in certain situations

### ISSUE THREAD

**Issue by weiznich** [dtolnay/trybuild#290]
I observe different outputs for [diesels compile tests](https://github.com/diesel-rs/diesel/tree/master/diesel_compile_tests) depending on whether I have the following `~/.cargo/config.toml` file:

```toml
[target.]
linker = "clang"
rustflags = ["-C", "link-arg=-fuse-ld=/usr/bin/mold"]
```

If the file exists it seems like `trybuild` does not pass the `--verbose` flag to the compiler, if it doesn't exist it will pass that flag, which results in widely different output. 

That's with `trybuild` 1.0.100 and rust 1.82.0 using the x86_64-unknown-linux-gnu target

**Comment by dtolnay** [comment]
I was able to reproduce this and observed that it's fixed by #293. Please let me know if those tests still fail on your end.

**Comment by weiznich** [comment]
Thanks for the fast fix :heart: 

I can confirm that this now works for me.
